crates/bevy_winit/src/accessibility.rs

```diff
@@ -0,0 +1,195 @@
+    mut receivers: ResMut<Handlers>,
+    mut events: EventReader<WindowClosed>,
+) {
+    for WindowClosed { id, .. } in events.iter() {
+        adapters.remove(id);
+        receivers.remove(id);
+    }
+}
+
+fn poll_receivers(handlers: Res<Handlers>, mut actions: EventWriter<ActionRequest>) {
+    for (_id, handler) in handlers.iter() {
+        let mut handler = handler.lock().unwrap();
+        while let Some(event) = handler.pop_front() {
+            actions.send(event);
+        }
+    }
+}
+
+fn update_accessibility_nodes(
+    adapters: NonSend<Adapters>,
+    query: Query<(Entity, &AccessibilityNode), Changed<AccessibilityNode>>,
+) {
+    let mut nodes = vec![];
+    for (entity, node) in &query {
+        nodes.push((entity.to_node_id(), Arc::new((**node).clone())));
+    }
+    if let Some(adapter) = adapters.get_primary_adapter() {
+        if !nodes.is_empty() {
+            let root_id = NodeId(NonZeroU128::new(WindowId::primary().as_u128()).unwrap());
+            let children = nodes.iter().map(|v| v.0).collect::<Vec<NodeId>>();
+            let window_update = (
+                root_id,
+                Arc::new(Node {
+                    role: Role::Window,
+                    children,
+                    ..default()
+                }),
+            );
+            nodes.insert(0, window_update);
+            adapter.update(TreeUpdate { nodes, ..default() });
+        }
+    }
+}
+
+fn remove_accessibility_nodes(
+    adapters: NonSend<Adapters>,
+    mut last_focus: ResMut<LastFocus>,
+    removed: RemovedComponents<AccessibilityNode>,
+    remaining_nodes: Query<Entity, With<AccessibilityNode>>,
+) {
+    if removed.iter().len() != 0 {
+        if let Some(last_focused_entity) = last_focus.entity() {
+            for entity in removed.iter() {
+                if entity == last_focused_entity {
+                    **last_focus = None;
+                    break;
+                }
+            }
+        }
+        if let Some(adapter) = adapters.get_primary_adapter() {
+            let root_id = NodeId(NonZeroU128::new(WindowId::primary().as_u128()).unwrap());
+            let children = remaining_nodes
+                .iter()
+                .map(|v| v.to_node_id())
+                .collect::<Vec<NodeId>>();
+            let window_update = (
+                root_id,
+                Arc::new(Node {
+                    role: Role::Window,
+                    children,
+                    ..default()
+                }),
+            );
+            let focus = (**last_focus).unwrap_or(root_id);
+            adapter.update(TreeUpdate {
+                nodes: vec![window_update],
+                focus: Some(focus),
+                ..default()
+            });
+        }
+    }
+}
+
+pub struct AccessibilityPlugin;
+
+impl Plugin for AccessibilityPlugin {
+    fn build(&self, app: &mut App) {
+        app.init_non_send_resource::<Adapters>()
+            .init_resource::<Handlers>()
```

Comment:
I think this was mentioned in a prior PR, but this is a very generic name. Could we bikeshed something a bit more localized for the plugin?